tmt/utils.py

```diff
@@ -2422,32 +2503,31 @@ def dataclass_field_by_name(cls: Any, name: str) -> 'dataclasses.Field[T]':
     retrieving a field when one knows its name.
 
     :param cls: a dataclass/data container class whose fields to search.
-    :param name: field name to retrieve.
+    :param key: field name to retrieve.
     :raises GeneralError: when the field does not exist.
     """
 
-    for field in dataclasses.fields(cls):
-        if field.name == name:
-            return field
-
-    else:
-        raise GeneralError(f"Could not find field '{name}' in class '{cls.__name__}'.")
+    for field in container_fields(container):
+        if field.name != key:
+            continue
 
+        metadata = field.metadata.get('tmt', FieldMetadata())
+        if inspect.isclass(container):
+            return field.name, key_to_option(field.name), field, metadata
 
-def dataclass_field_metadata(field: 'dataclasses.Field[T]') -> 'FieldMetadata[T]':
-    """
-    Return a dataclass/data container field metadata.
+        return (
+            field.name,
+            key_to_option(field.name),
+            container.__dict__[field.name],
+            field, metadata)
 
-    Dataclass fields have a mapping to hold fields' key/value metadata, and to
-    support linters in their job, instead of storing tmt's custom data directly
-    in the mapping, we use a special container to hold metadata we need.
+    from tmt.utils import GeneralError
```

Comment:
This is `tmt.utils` so no import needed?

Reply:
Nope, not necessary. Removed in 3ca9dab3.